### PR TITLE
feature/KAYA-51-Setup-Apollo-Client-For-HR-Panel

### PR DESCRIPTION
**Pull request by Kalpesh406**

### CODE REVIEW

File: hr-panel/.env.template [hunk 1]
```diff
@@ -1 +1,2 @@
-REACT_APP_AUTH_API_DOMAIN="http://localhost:8000/"
\ No newline at end of file
+REACT_APP_AUTH_API_DOMAIN="http://localhost:8000/"
```
Comment (eshwinrio) [inline hr-panel/.env.template]: Remove the quotes and the trailing slash after 8000.
```diff
- REACT_APP_AUTH_API_DOMAIN="http://localhost:8000/"
+ REACT_APP_AUTH_API_DOMAIN=http://localhost:8000 
```

Comment (eshwinrio) [reply hr-panel/.env.template]: Also add a new variable. Keep the variables sorted. Use VSCode to sort `Select all text> F1 > Sort Ascending`
```diff
+ REACT_APP_APPLICATION_SECRET=0f6b9c33-3293-4bc4-936e-5a6f94fd55ad
```

File: hr-panel/.env.template [hunk 2]
```diff
@@ -1 +1,2 @@
...
+REACT_APP_RESOURCE_API_DOMAIN="http://localhost:8001/"
```
Comment (eshwinrio) [inline hr-panel/.env.template]: Same updates to this variable
```diff
- REACT_APP_RESOURCE_API_DOMAIN="http://localhost:8001/"
+ REACT_APP_RESOURCE_API_DOMAIN=http://localhost:8001
```

File: hr-panel/codegen.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { CodegenConfig } from '@graphql-codegen/cli';
+
+const config: CodegenConfig = {
+  schema: process.env['REACT_APP_RESOURCE_API_DOMAIN'] + 'api/resource' ,
+  // this assumes that all your source files are in a top-level `src/` directory - you might need to adjust this to your file structure
```
Comment (eshwinrio) [inline hr-panel/codegen.ts]: Remove this single line comment

File: hr-panel/package.json [hunk 4]
```diff
@@ -14,8 +15,9 @@
     "@testing-library/user-event": "^13.5.0",
     "@types/jest": "^27.5.2",
     "@types/node": "^16.18.80",
-    "@types/react": "^18.2.55",
+    "@types/react": "^18.2.56",
     "@types/react-dom": "^18.2.19",
```
Comment (eshwinrio) [inline hr-panel/package.json]: Move these packages from `dependencies` to `devDependencies` as these packages should not be included in deployment builds.
```diff
- "@types/jest": "^27.5.2",
- "@types/node": "^16.18.80",
- "@types/react": "^18.2.56",
- "@types/react-dom": "^18.2.19",
```

File: hr-panel/package.json [hunk 5]
```diff
@@ -50,6 +52,9 @@
     ]
   },
   "devDependencies": {
+    "@graphql-codegen/cli": "^5.0.2",
+    "@graphql-codegen/client-preset": "^4.2.2",
+    "@graphql-typed-document-node/core": "^3.2.0",
     "@types/validator": "^13.11.9"
```
Comment (eshwinrio) [inline hr-panel/package.json]: Move those type definition packages here
```diff
+ "@types/jest": "^27.5.2",
+ "@types/node": "^16.18.80",
+ "@types/react": "^18.2.56",
+ "@types/react-dom": "^18.2.19",
```

**Review by eshwinrio — CHANGES_REQUESTED** [review]
Please update `hr-panel/src/lib/fetch-requests.ts` file to include new security headers along with those 2 already existing requests.
```diff
 export const verifyIdentity = (requestInit?: RequestInit) => fetch(
   `${process.env['REACT_APP_AUTH_API_DOMAIN']}/auth/verify`,
-  { credentials: 'include', ...requestInit }
+  {
+    method: 'GET',
+    headers: {
+      'X-Application': process.env['REACT_APP_APPLICATION_SECRET']!,
+    },
+    credentials: 'include',
+    ...requestInit
+  }
 );
 
-export const fetchAccessToken = (username: string, password: string) => fetch(
+export const fetchAccessToken = (username: string, password: string, requestInit?: RequestInit) => fetch(
   `${process.env['REACT_APP_AUTH_API_DOMAIN']}/auth/token`,
   {
     method: 'POST',
     headers: {
-      'Content-Type': 'application/json'
+      'Content-Type': 'application/json',
+      'X-Application': process.env['REACT_APP_APPLICATION_SECRET']!,
     },
     credentials: 'include',
     body: JSON.stringify({ username, password }),
+    ...requestInit
   }
 );
```

**Comment by Kalpesh406** [comment]
All the changes are done.

**Review by eshwinrio — APPROVED** [review]

**Pull request merged**